### PR TITLE
Convert binary operation normalizations to new normalization interface

### PR DESCRIPTION
**Pull request by phate**

### CODE REVIEW

**Review by haved — APPROVED** [review]
Looks good, I assume the added test BinaryOperator is not necessarily supposed to mimic any real world operator, just for testing.

File: tests/jlm/rvsdg/test-binary.cpp [hunk 1]
```diff
@@ -7,10 +7,80 @@
 #include "test-registry.hpp"
 #include "test-types.hpp"
 
+#include <jlm/rvsdg/NodeNormalization.hpp>
 #include <jlm/rvsdg/view.hpp>
 
-static void
-test_flattened_binary_reduction()
+class BinaryOperation final : public jlm::rvsdg::binary_op
```
[the remark this answers is not in the document]
Comment (phate) [reply tests/jlm/rvsdg/test-binary.cpp:13]: Yes, this was done deliberately. I am aware of 'jlm::tests::binary_op' and that I duplicate a lot of code here. It needs to be cleaned up eventually. The 'jlm::tests::binary_op' was not doing the job as I needed so implementations for 'can_reduce_operand_pair' etc.

**Comment by phate** [comment]
@haved Do you want me to change something, or can I merge?

**Comment by haved** [comment]
@phate I have marked my comment as resolved, merge away!

**Pull request merged**